src/main/java/com/conveyal/r5/profile/McRaptorSuboptimalPathProfileRouter.java

```diff
@@ -207,6 +213,10 @@ private boolean doOneRound () {
                 for (Map.Entry<StatePatternKey, McRaptorState> e : statesPerPatternSequence.entrySet()) {
                     int trip = tripsPerPatternSequence.get(e.getKey());
                     TripSchedule sched = pattern.tripSchedules.get(trip);
+                    //Skips trips which don't run on wanted date
+                    if(!servicesActive.get(sched.serviceCode)) {
+                        continue;
+                    }
```

Comment:
We don't need this check since you've added a check before we put the trips into tripPerPatternSequence.